File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.js
/**
 * @param {string} sentence
 * @param {string} searchWord
 * @return {number}
 */
var isPrefixOfWord = function(sentence, searchWord) {
    const words = sentence.split(" ");
    let n = words.length;

    for (let i = 0; i < n; i++){
        if(words[i].startsWith(searchWord)){
            return i + 1;
        }
    }

    return -1;
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        n = len(searchWord)
        lst = list(sentence.split())
        ans = -1

        for i in range(len(lst)):
            check = lst[i]
            if check.find(searchWord) != -1 and check[0] == searchWord[0] and check[n - 1] == searchWord[n - 1]:
                ans = i + 1
                break

        return ans